app/services/payment_providers/adyen_service.rb

```diff
@@ -64,6 +64,23 @@ def handle_event(organization:, event_json:)
 
         result = service.preauthorise(organization, event)
         result.raise_if_error! || result
+      when 'REFUND'
+        service = CreditNotes::Refunds::AdyenService.new
+
+        provider_refund_id = event["pspReference"]
+        status = event["success"] == "true" ? :succeeded : :failed
```

Comment:
Style/TernaryParentheses: Use parentheses for ternary expressions with complex conditions.